sympy/polys/series/tests/test_powerseriesring.py

```diff
@@ -0,0 +1,486 @@
+        R10,
+        R10.integrate(R10.from_list([2, 3, 4], 3)),
+        [(0, 1), (2, 1), (3, 2), (4, 3)],
+        4,
+    )
+
+
+def test_error(rd_int):
+    R, *_, r = rd_int
+    raises(ValueError, lambda: r(-1))
+    raises(ValueError, lambda: R.pow_int(R.gen, -1))
+    raises(ValueError, lambda: R.truncate(R.gen, -1))
+
+
+def test_int_compose(rd_int):
+    R, x, one, R3, x3, one3, R10, x10, one10, _ = rd_int
+
+    assert same(R, R.compose(R.from_list([2]), R.from_list([0, 1, 1])), [2], None)
+    assert same(R, R.compose(R.add(one, x), x), [1, 1], None)
+    assert same(R, R.compose(R.multiply(x, x), R.add(one, x)), [1, 2, 1], None)
+    assert same(
+        R, R.compose(R.from_list([1, 1, 1]), R.square(x)), [1, 0, 1, 0, 1], None
+    )
+    assert same(
+        R3, R3.compose(R3.from_list([1, 2, 3]), R3.from_list([0, 1, 1])), [1, 2, 5], 3
+    )
+    assert same(
+        R10,
+        R10.compose(
+            R10.from_list([2, 4, 5, 1, 6, 2], 7), R10.from_list([0, 1, 1, 2, 3, 4], 7)
+        ),
+        [2, 4, 9, 19, 46, 101, 206],
+        7,
+    )
+
+    raises(
+        ValueError, lambda: R.compose(R.from_list([1, 2], 2), R.from_list([1, 2, 3], 2))
+    )
+
+
+def test_rational_compose(rd_rational):
+    R, _, _, R3, _, _, R10, *_ = rd_rational
+
+    f1 = R.from_list([QQ(1, 2), QQ(3, 4)])
+    g1 = R.from_list([QQ(1, 3), QQ(2, 5)])
+    assert same(R, R.compose(f1, g1), [(3, 4), (3, 10)], None)
+
+    f3 = R.from_list([QQ(2, 3), QQ(5, 7)])
+    g3 = R.from_list([QQ(0, 1), QQ(3, 4), QQ(1, 6)])
+    assert same(R, R.compose(f3, g3), [(2, 3), (15, 28), (5, 42)], None)
+
+    f3_2 = R3.from_list([QQ(1, 4), QQ(1, 2), QQ(1, 8)])
+    g3_2 = R3.from_list([QQ(0, 1), QQ(1, 3)])
+    assert same(R3, R3.compose(f3_2, g3_2), [(1, 4), (1, 6), (1, 72)], 3)
+
+    f10 = R10.from_list([QQ(1, 2), QQ(3, 4), QQ(5, 6)], 4)
+    g10 = R10.from_list([QQ(0, 1), QQ(2, 3), QQ(1, 5), QQ(-3, 7)], 4)
+    assert same(R10, R10.compose(f10, g10), [(1, 2), (1, 2), (281, 540), (-25, 252)], 4)
+
+
+def test_int_inversion(rd_int):
+    R, x, one, R3, x3, one3, R10, x10, one10, _ = rd_int
+
+    assert same(R, R.inversion(R.add(one, x)), [1, -1, 1, -1, 1, -1], 6)
+    assert same(
+        R,
+        R.inversion(R.add(R.pow_int(R.multiply(R.add(one, x), x), 3), one)),
+        [1, 0, 0, -1, -3, -3],
+        6,
+    )
+    assert same(R3, R3.inversion(R3.from_list([1, 3, -2])), [1, -3, 11], 3)
+    assert same(
+        R10,
+        R10.inversion(R10.from_list([1, -2, 3])),
+        [1, 2, 1, -4, -11, -10, 13, 56, 73, -22],
+        10,
+    )
+
+    raises(NotReversible, lambda: R.inversion(R.zero))
+    raises(NotReversible, lambda: R.inversion(R.from_list([0, 1, 2])))
+
+
+def test_rational_inversion(rd_rational):
+    R, _, _, R3, _, _, R10, *_ = rd_rational
+
+    f_r = R.from_list([QQ(2, 3), QQ(-1, 4), QQ(3, 5)])
+    assert same(
+        R,
+        R.inversion(f_r),
+        [
+            (3, 2),
+            (9, 16),
+            (-729, 640),
```

Comment:
I don't like using the name "inversion" for this as it is used also for what is otherwise called "reversion" here. See e.g.:
https://en.wikipedia.org/wiki/Lagrange_inversion_theorem

I would call this (multiplicative) "inverse". The array API calls it "reciprocal" which is reasonable as a name though.